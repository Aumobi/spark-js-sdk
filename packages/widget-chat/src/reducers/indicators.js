import {ADD_TYPING_INDICATOR, DELETE_TYPING_INDICATOR} from '../actions/indicators';

export default function indicators(state = {
  typing: []
}, action) {
  switch (action.type) {
  case ADD_TYPING_INDICATOR:
    if (state.typing.indexOf(action.id) === -1) {
      return {typing: [action.id, ...state.typing]};
    }
    return state;
  case DELETE_TYPING_INDICATOR:
    return {
      typing: state.typing.filter((id) => id !== action.id)
    };
  default:
    return state;
  }
}
