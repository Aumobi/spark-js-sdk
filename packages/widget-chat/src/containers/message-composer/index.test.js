import React from 'react';
import renderer from 'react-test-renderer';

import {Provider} from 'react-redux';
import {applyMiddleware, combineReducers, compose, createStore} from 'redux';
import thunk from 'redux-thunk';

import {user, conversation, message} from '../../reducers';
import sparkReducer from '../../modules/redux-spark/reducers';


import MessageComposer from '.';

describe(`MessageComposer component`, () => {
  const store = createStore(
    combineReducers({
      user,
      conversation,
      message,
      spark: sparkReducer
    }),
    compose(applyMiddleware(thunk))
  );

  const component = renderer.create(
    <Provider store={store}>
      <MessageComposer
        placeholder="Message Placeholder"
        value="This is a message"
      />
    </Provider>
  );

  it(`renders properly`, () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it(`sends message properly`, () => {});

  it(`handles value change properly`, () => {
    const value = `abc123`;
    const event = {
      target: {
        value
      }
    };
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    const textarea = tree.children[0];
    textarea.props.onChange(event);
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
