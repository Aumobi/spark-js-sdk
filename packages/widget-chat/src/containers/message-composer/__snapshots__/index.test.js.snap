exports[`MessageComposer component handles value change properly 1`] = `
<div
  className="message-composer">
  <textarea
    className="textarea"
    onChange={[Function bound handleChange]}
    onKeyDown={[Function bound handleKeyDown]}
    onSubmit={[Function handleSubmit]}
    placeholder="Message Placeholder"
    value={undefined} />
</div>
`;

exports[`MessageComposer component handles value change properly 2`] = `
<div
  className="message-composer">
  <textarea
    className="textarea"
    onChange={[Function bound handleChange]}
    onKeyDown={[Function bound handleKeyDown]}
    onSubmit={[Function handleSubmit]}
    placeholder="Message Placeholder"
    value="abc123" />
</div>
`;

exports[`MessageComposer component renders properly 1`] = `
<div
  className="message-composer">
  <textarea
    className="textarea"
    onChange={[Function bound handleChange]}
    onKeyDown={[Function bound handleKeyDown]}
    onSubmit={[Function handleSubmit]}
    placeholder="Message Placeholder"
    value={undefined} />
</div>
`;
